src/resolve/resolved-mdns.c

```diff
@@ -386,8 +386,11 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
         if (r <= 0)
                 return r;
 
-        if (manager_packet_from_local_address(m, p))
-                return 0;
+        if (manager_packet_from_local_address(m, p)) {
+                /* if it's not a reply, handle this request */
+                if (dns_packet_validate_reply(p) > 0)
+                        return 0;
+        }
```

Comment:
so, this check is protection against accidentally answering our own packets, and getting confused by that and into possible reply loops. hence the check should really be kept in some form, but made tighter: we want to allow local stuff that is done via the classic unicast query format, but ignore it when doing mcast queries. that would allow your questions through that you were using in the reported issue.

to make such a check we just need to check the source UDP port nr, as per RFC 6762 Section 5.1:

```c
/* Refuse traffic from the local host, to avoid query loops. However, allow legacy mDNS unicast queries through anyway (we never send those ourselves, hence no risk). i.e. check for the source port nr. */
if (p->sender_port == MDNS_PORT && manager_packet_from_local_address(m, p)) 
        return 0;
```

Reply:
`mquery` supports known-answer suppression and things like that so it's not a legacy unicast querier. It needs fixing in some places like https://github.com/troglobit/mdnsd/issues/79 but the source port is correct. `resolved` should probably support mDNS queriers like this properly.

Reply:
Speaking of mDNS queriers resolved can't resolve its own local services either: https://github.com/systemd/systemd/pull/22532 and it seems that it should be properly supported too.